lib/util/git/behind-base-branch-cache.ts

```diff
@@ -0,0 +1,18 @@
+import { getCache } from '../cache/repository';
+
+export function getCachedBehindBaseResult(
+  branchName: string,
+  currentBaseBranchSha: string
+): boolean | null {
+  const cache = getCache();
+  const { branches = [] } = cache;
+  const cachedBranch = branches?.find(
+    (branch) => branch.branchName === branchName
+  );
+
+  if (!cachedBranch) {
+    return null;
+  }
+
+  return !(currentBaseBranchSha === cachedBranch.parentSha);
```

Comment:
How does this differ to a plain !== ?

Reply:
Is parentSha always written to cache at the right times?

Reply:
It is called each run I think
https://github.com/renovatebot/renovate/blob/d0926c3d6140436758ca18bb99708568fbacf553/lib/workers/repository/index.ts#L53

Reply:
It doesn't work as expected though, I tested it locally like this:
Create a new repo -> run renovate ->  update the baseBranch & add rebaseWhen="behind-base-branch" -> run renovate
After the second run finished I checked the cache but the parentSha and commit for the rebased branch were not updated in the cache.

Reply:
thhttps://github.com/renovatebot/renovate/blob/d0926c3d6140436758ca18bb99708568fbacf553/lib/workers/repository/update/branch/index.ts#L479

After commiting files to branch we can check if it null or a string. Incase it is a string we should update the sha of the branch in LocalConfig, this will make sure we have updated commits stored.

Reply:
Tested it locally like this:
1. Create a new repo with `rebaseWhen=behind-base-branch`
2. Run renovate
3. Update baseBranch
4. Run renovate and log the cache before and after branches were updated
Result: the cache values ie. parentSha and sha were updated.